subsys/usb/usb_c/usbc_pe_common_internal.h

```diff
@@ -302,6 +362,17 @@ bool policy_wait_notify(const struct device *dev, const enum usbc_policy_wait_t
  */
 void policy_set_src_cap(const struct device *dev, const uint32_t *pdos, const int num_pdos);
 
+/**
+ * @brief Check if the sink request can be met by the DPM
+ */
+enum usbc_snk_req_reply_t policy_check_sink_request(const struct device *dev,
+						const uint32_t request_msg);
+
+/**
+ * Check if the Present Contract is still valid
```

Comment:
After having now re-read this part of the spec a few times, we should probably include more details here for developer reference.  The contract is considered "invalid" if the previous current/voltage are no longer available AND the sink fails to make a valid request.  

This seems like something the PE "could" take care of for the DPM, but I'm not sure if we want to or not.  The spec doesn't mention that the DPM has to be the arbiter of this determination, but the DPM is tracking the source caps generally.